packages/taquito-sapling/src/sapling-forger/sapling-forger.ts

```diff
@@ -0,0 +1,81 @@
+import { SpendDescription, OutputDescription } from '../sapling-state/interface';
+import { toHexBuf, buf2hex } from '@taquito/utils';
+
+export class SaplingForger {
+  /**
+   * @description Forge sapling transactions
+   * @param spendDescriptions the list of spend descriptions
+   * @param outputDescriptions the list of output descriptions
+   * @param signature signature hash
+   * @param balance balance of the Sapling contract (input/output difference)
+   * @param root root of the merkle tree
+   * @returns Forged sapling transaction of type Buffer
+   */
+  forgeSaplingTransaction(
+    spendDescriptions: SpendDescription[],
+    outputDescriptions: OutputDescription[],
+    signature: Buffer,
+    balance: Buffer,
+    root: Buffer
```

Comment:
I think there is also a `bound_data` needed (it is a fix in jakarta).

In this shielded transaction:
00000000000001f3//7b1948a11abdcc241e362ddb5df01203ddfd1de5c24034e067ccd43006dc6706//b4d81b58638d535275111ab1df4e0289248c7510bc8c09e96f73cc446bda300d17455230448c4d67792358b8903bfbbca2135ca487cb447025ac8abbb2f4913f12cba1b7a72b601af99702336f54e1d95749968c34cc84eaf41c8c1ad77159a10dec8091ad5d8c7d17eec384886c204646a106d69895b10f26d4ff7ae57e8f8cc362983fc19c47f00d235e3d3abc7561838e749c54622f445e074d59091447270b8c555d3568f437fa95c02b71f028ddc364f2f78d63d287e28f74c554eb310c//0cf3d9812a872739f6d4eb9ecf56fade87fb95142604e0f6a2ac4c4a1c20fea1//fd99fea447d4dc1abcedd2784e301202cd560b1f6cdb5c17f792326448bbac30//0000004f//4f51c8ffcc0dfca5682bed933b2106f63f3d3205106f9bfa607524dc77c5283d1c24e9ee14de1ba20a85cad74b3fb2a4b45f564f2f690a953add2e3264712c83d764548c1d6e8eabd9fda9ee0821f9//72bdca1b7ce85d74a39df15a72b1e139e3e418e58373a6a0//81896d9eae230477b8dbef8a0b5f1b4b553a43a6e093e3daaa9950ff87c3bef51985abc80a5977edfe931d8b892c044f07f3caea99b9597bdd5cada8123b3047e38fc0197de15d87d042bdd0002970f7//7b3133016462818146ab850d9079a0c3defc970034eb8982//979b35e0a7756316fa76063ebe6f2c3d93bf10050ad80f0efb76c80f6f08ccc9dbd399f88fa0789ecfecf7f5efc5bd203ca4190938451100946478dd64cba202//ffffffffff676980//fbc2f4300c01f0b7820d00e3347c8da4ee614674376cbc45359daa54f9b5493e//00000000
The `00000000` at the end correspond to the bound_data.
It will only has a value (different than 00000000) for the unshield transaction

Reply:
is `bound_data` of type Buffer as well?

Reply:
yes